src/utils/algolia-queries.js

```diff
@@ -0,0 +1,51 @@
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const escapeStringRegexp = require('escape-string-regexp')
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+require('dotenv').config()
+
+const pagePath = `content/blog`
+const indexName = process.env.GATSBY_ALGOLIA_INDEX_NAME || 'dev_blogs'
+
+const pageQuery = `{
+  pages: allMarkdownRemark(
+    filter: {
+      fileAbsolutePath: { regex: "/${escapeStringRegexp(pagePath)}/" },
+    }
+  ) {
+    edges {
+      node {
+        id
+        frontmatter {
+          title
+          description
+        }
+        childBlogPost {
+          slug
+        }
+        excerpt(pruneLength: 1000)
+      }
+    }
+  }
+}`
```

Comment:
I think it'd be better to query on our custom `BlogPost` nodes, as the process of making those notes had already done the work of sorting out which `MarkdownRemark` nodes are blog posts as opposed to docs instead of re-doing that work here with a regex filter as they're known to be [particularly slow and less scalable than other methods of filtering](https://www.gatsbyjs.com/docs/query-filters/#performance).

Reply:
Yeah, we could use that approach. The issue was we couldn't find all the fields but it looks like we could use the inline query over the parent for those missing fields. So, I will consider this approach.